packages/edit-navigation/src/components/menus-editor/create-menu-panel.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { some } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	Panel,
+	PanelBody,
+	TextControl,
+	withFocusReturn,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+const noticeId = 'edit-navigation-create-menu-error';
+
+const menuNameMatches = ( menuName ) => ( menu ) =>
+	menu.name.toLowerCase() === menuName.toLowerCase();
+
+export function CreateMenuForm( { onCancel, menus } ) {
+	const [ menuName, setMenuName ] = useState( '' );
+	const { isCreatingMenu, hasCreatedMenu } = useSelect( ( select ) => {
+		const { getIsResolving, hasFinishedResolution } = select( 'core' );
+		return {
+			isCreatingMenu: getIsResolving( 'saveMenu' ),
+			hasCreatedMenu: hasFinishedResolution( 'saveMenu' ),
+		};
+	}, [] );
+	const { saveMenu } = useDispatch( 'core' );
+	const { createInfoNotice, createErrorNotice, removeNotice } = useDispatch(
+		'core/notices'
+	);
+
+	useEffect( () => {
+		if ( hasCreatedMenu ) {
+			createInfoNotice( __( 'Menu created' ), {
+				type: 'snackbar',
+				isDismissible: true,
+			} );
+		}
+	}, [ hasCreatedMenu ] );
```

Comment:
This doesn't seem to work. Maybe the component is unmounted before the effect gets a chance to fire.